infer/src/checkers/AbstractDomain.ml

```diff
@@ -87,6 +87,38 @@ module Pair (Domain1 : S) (Domain2 : S) = struct
     F.fprintf fmt "(%a, %a)" Domain1.pp astate1 Domain2.pp astate2
 end
 
+module Pair3 (Domain1 : S) (Domain2 : S) (Domain3 : S) = struct
```

Comment:
Wondering if we need this. Two invocations of the `Pair` functor can achieve the same thing, and for domains with > 2 components I think it's typically cleaner to use a record type for `astate` rather than a tuple. The advantage of using a record is that it's much easier to extend than a pair, and it's easier to ignore parts of the domain you don't care about without writing a bunch of `_`'s.